lib/oci_utils/impl/oci-image-migrate-import.py

```diff
@@ -0,0 +1,240 @@
+        console_msg('running: %s' % cmd)
+        bucketcontents = migrate_tools.run_popen_cmd(cmd)
+        if bucketcontents is not None:
+            bucketcontentsdict = json.loads(bucketcontents)
+        else:
+            console_msg(msg='bucket %s is emtpy.' % bucket)
+        #print('%s' % bucketcontentsdict)
+    except Exception as e:
+        exit_with_msg('Failed to get contents of %s: %s' % (bucket, str(e)))    
+    #
+    # object exists?
+    if migrate_utils.object_exists(bucketcontents, objectname):
+        console_msg(msg='Object %s present in %s.' % (objectname, bucket))
+    else:
+        exit_with_msg('Object %s missing from %s.' % (objectname, bucket))
+    #
+    # display name exists?
+    cmd = ['oci', 'compute', 'image', 'list', '--compartment-id',
+           '%s' % compartment_id, '--display-name',
+           '%s' % displayname]
+    objstat = migrate_tools.run_popen_cmd(cmd)
+    if objstat is not None:
+        exit_with_msg('Image with name %s already exists.' % displayname)
+    #
+    # upload?
+    if not read_yn('\n  Uploading %s to %s as %s'
+                   % (objectname, compartment, displayname)):
+        exit_with_msg('Exiting.\n')
+    cmd = ['oci', 'compute', 'image', 'import', 'from-object', '--bucket-name',
+           '%s' % bucket, '--compartment-id', '%s' % compartment_id,
+           '--name', '%s' % objectname, '--namespace', '%s' % namespace,
+           '--launch-mode',  launchmode, '--display-name', '%s' % displayname]
+    console_msg(msg='running: %s' % cmd)
+    try:
+        resval = migrate_tools.run_popen_cmd(cmd)
+        #console_msg(msg='running import: %s' % resval)
+    except Exception as e:
+        exit_with_msg('Failed to import %s: %s' % (objectname, str(e)))
+    #
+    # follow up
+    cmd = ['oci', 'compute', 'image', 'list', '--compartment-id',
+           '%s' % compartment_id, '--display-name', '%s' % displayname]
+    # console_msg(msg='running: %s' % cmd)
+    finished = False
+    year, month, day, hour, min = map(int, time.strftime("%Y %m %d %H %M").split())
+    sys.stdout.write('\n  %02d-%02d-%04d %02d:%02d %s\n'
+                     % (day, month, year, hour, min,
+                        'Starting import of %s' % displayname))
+    sys.stdout.flush()
+    cnt=0
+    modcnt=40
+    addn = -1
+    while not finished:
+        objstat = json.loads(migrate_tools.run_popen_cmd(cmd))
+        for ob in objstat['data']:
+            if ob['display-name'] == displayname:
+                if ob['lifecycle-state'] == 'AVAILABLE':
+                    finished = True
+                else:
+                    addn *= -1
+                    for i in range(modcnt-1):
+                        cntstr = (cnt%modcnt+1)*'.' + (modcnt-cnt%modcnt)*' '
+                        year, month, day, hour, min \
+                            = map(int, time.strftime("%Y %m %d %H %M").split())
+                        sys.stdout.write('  %02d-%02d-%04d %02d:%02d %s %s\r'
```

Comment:
Why not using your helpers like console_msg ?

Reply:
console message writes newline's , this message is intended to be a time-counter while importing the image, which can take really quite some time ...